lib/db/memory.js

```diff
@@ -260,27 +250,39 @@ MemoryStore.prototype = {
     }
 
     var activeClientIds = [];
-    var activeClients = [];
+    var activeClients = {};
     var ids = Object.keys(this.tokens);
     for (var i = 0; i < ids.length; i++) {
       var id = ids[i];
       if (this.tokens[id].userId.toString('hex') === uid) {
-        activeClientIds.push(this.tokens[id].clientId);
+        activeClientIds.push({
+          clientId: this.tokens[id].clientId,
+          clientTokenTime: this.tokens[id].createdAt
+        });
       }
     }
 
     // unique clients
-    activeClientIds.forEach(function (clientIdBuf) {
-      var clientIdHex = unbuf(clientIdBuf);
-      if (! findByBufProp(activeClients, 'id', clientIdHex)) {
-        var c = self.clients[clientIdHex];
-        if (c.canGrant === false) {
-          activeClients.push(c);
+    activeClientIds.forEach(function (clientBuf) {
```

Comment:
The name `clientIdBuf` was to help point out that it was a `Buffer` of the client ID. Seems that it no longer is, so it could probably be renamed to just `client` or similar.